lib/DirectEditing.js

```diff
@@ -162,7 +162,7 @@ DirectEditing.prototype.activate = function(element) {
   var context;
 
   var provider = find(this._providers, function(p) {
-    return (context = p.activate(element)) ? p : null;
+    return ((context = p.activate(element))) ? p : null;
```

Comment:
Is that considerably better? I don't know. It is still an anti pattern to extract `context` like that. I'd say: Let's try to forget about that line of code instead of making it more verbose :see_no_evil:.

Reply:
True, the whole thing is difficult to grasp 🙈 .